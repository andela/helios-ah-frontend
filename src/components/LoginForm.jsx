import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import TextFieldGroup from './TextFieldGroup';
import FlashMessageList from './FlashMessagesList';
import SubmitButton from './SubmitButton';
import { baseUrl } from '../utilities/apiRequests';

class LoginForm extends Component {
  platformRedirect = platform => this.openLink(`${baseUrl}/auth/${platform}`);

  render() {
    return (
      <div className="container login-body">
        <div className="row">
          <div className="col-md-6">
            <span className="login-image-container" />
          </div>
          <div className="col-md-6 col-xs-12 right-login">
            <h3>WELCOME BACK.....</h3>
            <h5>LOGIN TO AUTHOR`S HAVEN</h5>

            <FlashMessageList customAlertClass="login-custom-alert" />

            <form autoComplete="off">
              <TextFieldGroup
                error={this.props.errors && this.props.errors.email}
                placeholder="Email"
                field="email"
                value={this.props.email}
                onChange={this.props.onChange}
                onBlur={this.props.onBlur}
                errorFeedbackClass="login-invalid-feedback"
                validFeedbackClass="login-valid-feedback"
                onInput={this.props.onInput}
                customFormDivClass="login-div-class"
                className="form-control login-input"
              />
              <TextFieldGroup
                error={this.props.errors && this.props.errors.password}
                placeholder="Password"
                field="password"
                type="password"
                value={this.props.password}
                onChange={this.props.onChange}
                onBlur={this.props.onBlur}
                errorFeedbackClass="login-invalid-feedback"
                validFeedbackClass="login-valid-feedback"
                onInput={this.props.onInput}
                customFormDivClass="login-div-class"
                className="form-control login-input"
              />
              <div className="forgot-password">
                <Link to="/reset-password">
                  <p>FORGOT PASSWORD?</p>
                </Link>
              </div>
              <div className="login-buttons">
                <SubmitButton
                  onClick={this.props.submitDetails}
                  isRequestSent={this.props.isRequestSent}
                  className="login-btn"
                  value="LOGIN"
                  columnAttribute="login-col"
                />
              </div>

              <Link to="/signup" className="signup-link-container">
                <div className="signup-link">Unregistered? Sign up here</div>
              </Link>

              <div className="social-media-icons">
                <a href={`${baseUrl}/auth/social_fb`}>
                  <span
                    className="facebook"
                  />
                </a>
                <a href={`${baseUrl}/auth/social_ggl`}>
                  <span
                    className="google"
                  />
                </a>
                <a href={`${baseUrl}/auth/social_tw`}>
                  <span
                    className="twitter"
                  />
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

LoginForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onInput: PropTypes.func,
  onBlur: PropTypes.func,
  errors: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string)
  ]).isRequired,
  submitDetails: PropTypes.func.isRequired,
  isRequestSent: PropTypes.bool.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired
};

LoginForm.defaultProps = {
  onInput: () => null,
  onBlur: () => null
};

export default LoginForm;
