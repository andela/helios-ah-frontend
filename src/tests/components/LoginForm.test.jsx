import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../components/LoginForm';

describe('Render Login Page', () => {
  describe('render Login Page', () => {
    const propsObj = {
      onChange: () => {},
      onBlur: () => {},
      onInput: () => {},
      errors: 'some error',
      isRequestSent: true,
      submitDetails: () => {},
      email: 'some@email.com',
      password: 'password'
    };
    it('renders the login page component', () => {
      const wrapper = shallow(<LoginForm {...propsObj} />);
      expect(wrapper.find('div').length).toEqual(8);
      expect(wrapper.find('div').find('span').length).toEqual(4);
      expect(wrapper.find('TextFieldGroup').length).toEqual(2);
      expect(wrapper.find('SubmitButton').length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('h3').length).toEqual(1);
      expect(wrapper.find('h5').length).toEqual(1);
      expect(wrapper.find('form').length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
