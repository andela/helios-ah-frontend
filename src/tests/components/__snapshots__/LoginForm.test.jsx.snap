// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Page render Login Page renders the login page component 1`] = `
<div
  className="container login-body"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <span
        className="login-image-container"
      />
    </div>
    <div
      className="col-md-6 col-xs-12 right-login"
    >
      <h3>
        WELCOME BACK.....
      </h3>
      <h5>
        LOGIN TO AUTHOR\`S HAVEN
      </h5>
      <Connect(FlashMessageList)
        customAlertClass="login-custom-alert"
      />
      <form
        autoComplete="off"
      >
        <TextFieldGroup
          className="form-control login-input"
          customFormDivClass="login-div-class"
          error=""
          errorFeedbackClass="login-invalid-feedback"
          field="email"
          id={null}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Email"
          type="text"
          validFeedbackClass="login-valid-feedback"
          value="some@email.com"
        />
        <TextFieldGroup
          className="form-control login-input"
          customFormDivClass="login-div-class"
          error=""
          errorFeedbackClass="login-invalid-feedback"
          field="password"
          id={null}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Password"
          type="password"
          validFeedbackClass="login-valid-feedback"
          value="password"
        />
        <div
          className="forgot-password"
        >
          <Link
            replace={false}
            to="/reset-password"
          >
            <p>
              FORGOT PASSWORD?
            </p>
          </Link>
        </div>
        <div
          className="login-buttons"
        >
          <SubmitButton
            className="login-btn"
            columnAttribute="login-col"
            id={null}
            isRequestSent={true}
            onClick={[Function]}
            value="LOGIN"
          />
        </div>
        <Link
          className="signup-link-container"
          replace={false}
          to="/signup"
        >
          <div
            className="signup-link"
          >
            Unregistered? Sign up here
          </div>
        </Link>
        <div
          className="social-media-icons"
        >
          <a
            href="http://localhost:4001/api/v1/auth/social_fb"
          >
            <span
              className="facebook"
            />
          </a>
          <a
            href="http://localhost:4001/api/v1/auth/social_ggl"
          >
            <span
              className="google"
            />
          </a>
          <a
            href="http://localhost:4001/api/v1/auth/social_tw"
          >
            <span
              className="twitter"
            />
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Render Login Page render Login Page renders the login page component 2`] = `
<div
  className="container login-body"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <span
        className="login-image-container"
      />
    </div>
    <div
      className="col-md-6 col-xs-12 right-login"
    >
      <h3>
        WELCOME BACK.....
      </h3>
      <h5>
        LOGIN TO AUTHOR\`S HAVEN
      </h5>
      <Connect(FlashMessageList)
        customAlertClass="login-custom-alert"
      />
      <form
        autoComplete="off"
      >
        <TextFieldGroup
          className="form-control login-input"
          customFormDivClass="login-div-class"
          error=""
          errorFeedbackClass="login-invalid-feedback"
          field="email"
          id={null}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Email"
          type="text"
          validFeedbackClass="login-valid-feedback"
          value="some@email.com"
        />
        <TextFieldGroup
          className="form-control login-input"
          customFormDivClass="login-div-class"
          error=""
          errorFeedbackClass="login-invalid-feedback"
          field="password"
          id={null}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Password"
          type="password"
          validFeedbackClass="login-valid-feedback"
          value="password"
        />
        <div
          className="forgot-password"
        >
          <Link
            replace={false}
            to="/reset-password"
          >
            <p>
              FORGOT PASSWORD?
            </p>
          </Link>
        </div>
        <div
          className="login-buttons"
        >
          <SubmitButton
            className="login-btn"
            columnAttribute="login-col"
            id={null}
            isRequestSent={true}
            onClick={[Function]}
            value="LOGIN"
          />
        </div>
        <Link
          className="signup-link-container"
          replace={false}
          to="/signup"
        >
          <div
            className="signup-link"
          >
            Unregistered? Sign up here
          </div>
        </Link>
        <div
          className="social-media-icons"
        >
          <a
            href="http://localhost:4001/api/v1/auth/social_fb"
          >
            <span
              className="facebook"
            />
          </a>
          <a
            href="http://localhost:4001/api/v1/auth/social_ggl"
          >
            <span
              className="google"
            />
          </a>
          <a
            href="http://localhost:4001/api/v1/auth/social_tw"
          >
            <span
              className="twitter"
            />
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
